### PR TITLE
appliance/mongodb: Fix UserExists check

### LINKED ISSUE #3316: mongodb: Reports unhealthy
```
$ flynn resource add mongodb
Created resource b563fbb0-09e4-4f5d-b09d-c8f69d5b7dcc and release a4e457d4-9905-4244-b851-975138d89d0e.
```

```
$ flynn run curl -i mongodb.discoverd:27018/.well-known/status
HTTP/1.1 500 Internal Server Error
Content-Type: application/json
Date: Fri, 19 Aug 2016 21:34:56 GMT
Content-Length: 76

{
  "data": {
    "status": "unhealthy",
    "version": "v20160814.0"
  }
}
```

https://gist.github.com/9becbb975eb555be443dd54650916976

### PR DESCRIPTION
**Pull request by josephglanville**
Closes #3316

### CODE REVIEW

File: appliance/mongodb/process.go [hunk 1]
```diff
@@ -770,7 +770,7 @@ func (p *Process) userExists() (bool, error) {
 	if err := session.Run(bson.M{"usersInfo": bson.M{"user": "flynn", "db": "admin"}}, &entry); err != nil {
 		return false, err
 	}
-	return entry.Retval != nil, nil
+	return entry.Retval == nil, nil
```
Comment (lmars) [inline appliance/mongodb/process.go]: So it returns `nil` when you ask for the user info? That seems odd!

Comment (josephglanville) [inline appliance/mongodb/process.go]: Actually you might be right.. I am going to take a closer look at the query because something isn't quite right.

**Comment by josephglanville** [comment]
@lmars I think the problem was some of this was leftover from before mgo upgrade, I fixed this up so it actually parses the response.

File: bootstrap/manifest_template.json [hunk 2]
```diff
@@ -412,6 +412,7 @@
     "release": {
       "env": {
         "FLYNN_MONGO": "mongodb",
+        "MONGO_DATABASE": "admin",
```
Comment (lmars) [inline bootstrap/manifest_template.json]: Do we also need a migration to update existing mongo apps to have this env var?

Comment (josephglanville) [inline bootstrap/manifest_template.json]: Shouldn't be necessary, this is just a convenience when debugging and consistency with the other appliances. I don't think it's likely to be used by users as it is only useful for examining stuff like replica set configuration.

**Comment by titanous** [comment]
LGTM

**Pull request merged**